File: client/src/app/property/services/property.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../environments/environment.development';
import { PropertyFilter } from '../models/property-filter.model';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PropertiesResponse } from '../models/properties-response.model';
import { HttpParamsBuilder } from '../../util/http-params-builder';

@Injectable({ providedIn: 'root' })
export class PropertyService {
  private readonly URL = `${environment.serverUrl}/properties`;

  constructor(private http: HttpClient) {}

  public getProperties(
    page: number,
    size: number,
    propertyFilter: PropertyFilter
  ): Observable<PropertiesResponse> {
    const params = new HttpParamsBuilder(page, size, propertyFilter).build();
    return this.http.get<PropertiesResponse>(this.URL, { params: params });
  }
}

File: server/src/main/java/com/moh/property_reservation/property/service/PropertyService.java
package com.moh.property_reservation.property.service;

import org.springframework.data.domain.Pageable;

import com.moh.property_reservation.property.dto.PropertyFilter;
import com.moh.property_reservation.property.dto.PropertiesResponse;

public interface PropertyService {
    PropertiesResponse getProperties(PropertyFilter filter, Pageable pageable);
}
